bench/workgen/workgen.cxx

```diff
@@ -1106,13 +1105,40 @@ void Operation::kv_gen(bool iskey, uint32_t randomizer, uint64_t n,
     if (n > max)
         THROW((iskey ? "Key" : "Value") << " (" << n
           << ") too large for size (" << size << ")");
-    if (randomizer != 0) {
-        randomizer %= 1000;
-        snprintf(result, 6, ":%3.3d:", randomizer);
-        n -= RANDOMIZER_SIZE;
-        result += RANDOMIZER_SIZE;
-    }
+    /* Setup the buffer, defaulting to zero filled. */
     workgen_u64_to_string_zf(n, result, size);
+
+    /*
+     * Compressability is a percentage, 100 is all zeroes, it applies to the
+     * proportion of the value that can't be used for the identifier.
+     */
+    if (size > 20 && compressibility < 100) {
+        static const char alphanum[] =
+          "0123456789"
+          "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+          "abcdefghijklmnopqrstuvwxyz";
+        /*
+         * The random length is the proportion of the string that should not
+         * be compressible. As an example a compressibility of 25 in a value
+         * of length 100 should be:
+         * 100 - ((100 * 25) / 100) = 75
+         * That means that 75% of the string will be random numbers, and 25
+         * will be easily compressible zero-fill.
+         */
+        uint64_t random_len = size - ((size * compressibility) / 100);
+
+        /* Never overwrite the record number identifier */
+        if (random_len > size - 20)
+            random_len = size - 20;
+
+        for (int i = 0; i < random_len; ++i)
+            /*
+             * TODO: It'd be nice to use workgen_rand here, but this class
+             * is without the context of a runner thread, so it's not easy
+             * to get access to a state.
```

Comment:
@agorrod, to get the equivalent of calling workgen_rand here, I think maybe passing the ThreadRunner as an argument to kv_gen(), and then have a small method on ThreadRunner that returns a random value.

Reply:
I made a change like that - it was more invasive than I'd hoped, with a possible layering violation of referencing `ThreadRunner` in `workgen.h`